introduction_to_amazon_algorithms/jumpstart-foundation-models/retrieval_augmented_generation_question_answering_embedding.ipynb

```diff
@@ -0,0 +1,688 @@
+    "    deploy_image_uri = image_uris.retrieve(\n",
+    "        region=None,\n",
+    "        framework=None,  # automatically inferred from model_id\n",
+    "        image_scope=\"inference\",\n",
+    "        model_id=model_id,\n",
+    "        model_version=model_version,\n",
+    "        instance_type=inference_instance_type,\n",
+    "    )\n",
+    "    # Retrieve the model uri.\n",
+    "    model_uri = model_uris.retrieve(\n",
+    "        model_id=model_id, model_version=model_version, model_scope=\"inference\"\n",
+    "    )\n",
+    "    model_inference = Model(\n",
+    "        image_uri=deploy_image_uri,\n",
+    "        model_data=model_uri,\n",
+    "        role=aws_role,\n",
+    "        predictor_cls=Predictor,\n",
+    "        name=endpoint_name,\n",
+    "    )\n",
+    "    model_predictor_inference = model_inference.deploy(\n",
+    "        initial_instance_count=1,\n",
+    "        instance_type=inference_instance_type,\n",
+    "        predictor_cls=Predictor,\n",
+    "        endpoint_name=endpoint_name,\n",
+    "    )\n",
+    "    print(f\"{bold}Model {model_id} has been deployed successfully.{unbold}{newline}\")\n",
+    "    _MODEL_CONFIG_[model_id][\"endpoint_name\"] = endpoint_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2. Ask a question to **Flan T5 XXL**, **BloomZ 7b1**, and **Flan T5 UL2** without providing the Context\n",
+    "\n",
+    "To better illustrate why we need retrieval-augmented generation (RAG) based approach to solve the question and anwering problem. Let's directly ask the model a question and see how they respond."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "question = \"Which instances can I use with Managed Spot Training in SageMaker?\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "payload = {\n",
+    "    \"text_inputs\": question,\n",
+    "    \"max_length\": 100,\n",
+    "    \"num_return_sequences\": 1,\n",
+    "    \"top_k\": 50,\n",
+    "    \"top_p\": 0.95,\n",
+    "    \"do_sample\": True,\n",
+    "}\n",
+    "\n",
+    "\n",
+    "for model_id in _MODEL_CONFIG_:\n",
+    "    endpoint_name = _MODEL_CONFIG_[model_id][\"endpoint_name\"]\n",
+    "    query_response = query_endpoint_with_json_payload(\n",
+    "        json.dumps(payload).encode(\"utf-8\"), endpoint_name=endpoint_name\n",
+    "    )\n",
+    "    generated_texts = _MODEL_CONFIG_[model_id][\"parse_function\"](query_response)\n",
+    "    print(f\"For model: {model_id}, the generated output is: {generated_texts[0]}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can see all models generated the worng answer. The correct answer should be united states."
```

Comment:
The correct answer to \"Which instances can I use with Managed Spot Training in SageMaker?\"" is "united states"?!

Reply:
Sorry Laurent. Yesterday's PR is a demo. I have removed it in today's revision. And i will address your following comments. Thanks for reviewing.